### ISSUE TITLE
Slow response when using S3

### ISSUE THREAD

**Issue by oliverschewe** [thumbor/thumbor-aws#15]
My response time increases from 60ms to 1500ms when using `RESULT_STORAGE = 'thumbor_aws.result_storage'` compared to no result storage. Any idea why or how to debug this?
I am coming from thumbor==6 with python2.7 and wanted to upgrade to version thumbor v7, also switching from tc_aws to thumbor-aws. Everything works fine except for this really high response time.

Installation
```
FROM python:3.10-slim
...
thumbor==7.0.8
thumbor-aws==0.4.0
```

thumbor.conf
```
RESULT_STORAGE = 'thumbor_aws.result_storage'
RESULT_STORAGE_EXPIRATION_SECONDS = 0
RESULT_STORAGE_STORES_UNSAFE = True

### AWS Storage ###
AWS_RESULT_STORAGE_REGION_NAME='eu-central-1'
AWS_RESULT_STORAGE_BUCKET_NAME='thumbor-adsadaaasd'
AWS_RESULT_STORAGE_ROOT_PATH='rs'
AWS_DEFAULT_LOCATION='https://thumbor-adsadaaasd.s3.eu-central-1.amazonaws.com'
```

I am using /.aws/config to assume aws role for authorization.

Edit: I think it is due to the fact that the boto aws session is not being reused. It looks a new session (via AWS assumeRole) is created for every single request.

**Comment by guilhermef** [comment]
Hi @oliverschewe, we can probably reuse the existing client like it's suggested here: https://github.com/aio-libs/aiobotocore/issues/928

Right now I don't have a way to test it, do you think you can create a PR?

**Comment by guilhermef** [comment]
@oliverschewe I've changed the client creation to reuse the same instance, can you test it?

**Comment by guilhermef** [comment]
Hi @oliverschewe, thanks for testing it.
It seems that the initial request sets the expiration to `now`.
https://github.com/aio-libs/aiobotocore/blob/4694ba4761c0a6cc9a31c620f4c9eb9eff653bf1/aiobotocore/credentials.py#L693

But then, the other ones should have a longer expiration time.
How often do you see the token refresh on your logs?

**Comment by oliverschewe** [comment]
I don't see any refresh. It's like there never was a session a before, when I stress test my thumbor app with 10 sequential requests it simple takes 10x 1-2 seconds.

I created a simple app with aiobotocore to test if this has something to with its assume-role-with-web-identity functionality, but it looks ok. The session is only created once and reused in every subsequently use.
All credentials are read by botocore's provider pipeline (env, assume-role, assume-role-with-web-identity, sso, ...), so no need to pass any access_keys or secrets.
S3 is called every 60 seconds, only the first calls AssumeRoleWithWebIdentity. Strangely, shown expiration time from response is current time - but works nevertheless. (Sorry for the poor python, coming from Java)

```
import asyncio
import time
import logging
from aiobotocore.session import AioSession, get_session

AWS_SERVICE='s3'
AWS_REGION='eu-central-1'

BUCKET='my-thumbor-bucket'
BUCKET_KEY='rs/auto_webp/unsafe/400x301/filters:quality(80)/example.com/image.jpeg'


async def test_s3(aws_session):

    async with aws_session.create_client(service_name=AWS_SERVICE, region_name=AWS_REGION) as s3:

        response = await s3.list_buckets()
        logging.info(response['Buckets'][0])

        response = await s3.get_object(Bucket=BUCKET, Key=BUCKET_KEY)
        logging.info('File: {} {}'.format(response['ResponseMetadata']['HTTPStatusCode'], response['ContentLength']))

    time.sleep(60)
    await test_s3(aws_session)


async def start():
    session: AioSession = get_session()

    await test_s3(session)


if __name__ == '__main__':
    logging.basicConfig(level=logging.DEBUG)
    asyncio.run(start())

```

**Comment by guilhermef** [comment]
I have good news @oliverschewe. I've tracked down the problem.
Thumbor is recreating the Storage on every request, so we're reusing the client created on the previous request.
I'll try to address this issue.

**Comment by guilhermef** [comment]
@oliverschewe I've updated the PR, and tested it.
Now it's reusing the authentication.

**Comment by guilhermef** [comment]
The fix will be released on 0.4.3

**Comment by guilhermef** [comment]
https://pypi.org/project/thumbor-aws/0.4.3/
